server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -338,3 +341,86 @@ export class MetadataExtractionProcessor {
     return Duration.fromObject({ seconds: videoDurationInSecond }).toFormat('hh:mm:ss.SSS');
   }
 }
+
+@Processor(QueueName.SIDECAR)
+export class SidecarProcessor {
+  private logger = new Logger(SidecarProcessor.name);
+  private assetCore: AssetCore;
+
+  constructor(
+    @Inject(IAssetRepository) private assetRepository: IAssetRepository,
+    @Inject(IJobRepository) private jobRepository: IJobRepository,
+  ) {
+    this.assetCore = new AssetCore(assetRepository, jobRepository);
+    this.init();
+  }
+
+  private async init() {
+  }
+
+  @Process(JobName.QUEUE_SIDECAR)
+  async handleQueueSidecar(job: Job<IBaseJob>) {
+    try {
+      const { force } = job.data;
+      const assets = force ? await this.assetRepository.getWith(WithProperty.SIDECAR) : await this.assetRepository.getWithout(WithoutProperty.SIDECAR);
+
+      for (const asset of assets) {
+        const fileName = asset.originalFileName
+        const name = force ? JobName.SIDECAR_SYNC : JobName.SIDECAR_DISCOVERY;
+        await this.jobRepository.queue({ name, data: { asset, fileName } });
+      }
+    } catch (error: any) {
+      this.logger.error(`Unable to queue sidecar scanning`, error?.stack);
+    }
+  }
+
+  @Process(JobName.SIDECAR_DISCOVERY)
+  async handleQueueSidecarDiscovery(job: Job<IAssetUploadedJob>) {
+    let { asset, fileName } = job.data
+    if (!asset.isVisible) {
+      return;
+    }
+
+    try {
+      const sidecarPath = await this.getSidecarPath(asset);
+      if (sidecarPath) {
+        asset = await this.assetCore.save({ id: asset.id, sidecarPath });
+        const name = asset.type === AssetType.VIDEO ? JobName.EXTRACT_VIDEO_METADATA : JobName.EXIF_EXTRACTION;
+        await this.jobRepository.queue({ name, data: { asset, fileName } });
```

Comment:
the metadata job, the filename is coming from asset.originalFileName. That is semi new, but it makes this extra filename param obsolete. It can be removed everywhere, we just haven't gotten around to it yet, but you definitely don't need to use it for any new jobs.

Reply:
Unfortunately this seems required. The `JobName.EXIF_EXTRACTION` requires a parameter of `IAssetUploadedJob` and does not accept `IAssetJob`. Due to this, it's required I pass in the filename. Unless you want me to refactor this now, it is throwing transpiling errors.